### PR TITLE
Add editable solution block XML to all Blockly levels, not just artist, calc, and eval.

### PR DESCRIPTION
**Pull request by Hamms**
Formerly, most kinds of Blockly levels supported viewing solutions via a mechanism that scanned the DB for the most commonly-used LevelSource that successfully solved the level, and saving a reference to that as the "ideal level source,"

This had a couple of problems. Most notably, if the level were to ever change, if the process were to ever fail, or if the solution selected was correct but not ideal, it was difficult to change the solution.

This PR expands the concept of Levelbuilder-defined solutions to include all kinds of Blockly levels in an effort to fix these problems.

A companion PR (https://github.com/code-dot-org/code-dot-org/pull/6942) migrates the existing solutions from the ideal level sources.

### CODE REVIEW

**Comment by Hamms** [comment]
FYI @laurelfan

**Comment by breville** [comment]
Out of my usual territory but seems okay.  Did any tests need updating?

**Comment by Hamms** [comment]
Yeah, I wasn't entirely sure for whom this fell into usual territory. Is there anyone else you think should look this over?

I'll double-check the dashboard tests, but I'm reasonably confident that no tests currently look at solutions. Mostly because up til now, the data for solutions lived exclusively on production.

Creating tests for solutions and perhaps even creating a continuous deployment system reliant on those tests are part of the longer-term plan for this work.

**Comment by breville** [comment]
@joshlory usually knows a lot about these dashboard schemas :)

**Comment by joshlory** [comment]
Not as much as @laurelfan :)

What is the deprecation plan for "ideal level source"?

**Comment by Hamms** [comment]
After the content push goes through to update all levels with the new solution xml, I'll run another test to verify that the sets of valid/invalid levels haven't changed, then I'll go through and remove ideal level sources

File: dashboard/app/controllers/levels_controller.rb [hunk 1]
```diff
@@ -41,6 +41,17 @@ def edit_blocks
...
     blocks_xml = @level.properties[type].presence || @level[type] || EMPTY_XML
+
+    # The concept of ideal level sources predates the concept of
+    # solution blocks for all Blockly levels. A level could have an
+    # ideal level source but no solution blocks if it had an ideal level
+    # source before solution blocks were added; in that case, we want to
+    # use the ideal level source to define the initial set of solution
+    # blocks
+    if type == 'solution_blocks' && blocks_xml == EMPTY_XML && @level.ideal_level_source
+      blocks_xml = @level.ideal_level_source.data
```
Comment (laurelfan) [inline dashboard/app/controllers/levels_controller.rb]: have you checked that ideal_level_source is available on levelbuilder? I am not sure that it is.

Comment (Hamms) [inline dashboard/app/controllers/levels_controller.rb]: oh, good call, this is probably useless, and is more importantly made irrelevant by my addition of solution blocks en masse.

I'll remove this.

**Comment by laurelfan** [comment]
This could be a good opportunity to add test coverage for the solution block editing in LevelsController. I noticed this is not covered (at the controller level, it is at the model leve): https://github.com/code-dot-org/code-dot-org/blob/bdc96771c1de6d1933a7b708f7fdf5b1a907eae2/dashboard/app/controllers/levels_controller.rb#L70 should this functionality be moved to Blockly and not just Artist?

Otherwise LGTM.

**Comment by Hamms** [comment]
Yep, I plan to do a separate PR that expands the auto-population stuff to all Blockly and adds some tests (and potentially some more auto-population / verification stuff)

**Pull request merged**